megamek/src/megamek/common/building/BuildingSection.java

```diff
@@ -0,0 +1,294 @@
+    private int armor;     // >= 0
+    private boolean basementCollapsed;
+    private boolean burning;
+
+    /**
+     * @return the coordinates of this building section
+     */
+    public Coords getCoordinates() {
+        return coordinates;
+    }
+
+   /**
+    * @return the amount of damage the building absorbs
+    */
+   public int getAbsorbtion() {
+       return (int) Math.ceil(phaseCF / 10.0);
+   }
+
+   /**
+    * @return the basement type of this building section
+    */
+    public BasementType getBasementType() {
+        return basementType;
+    }
+
+    /**
+     * Sets this building section's basement type to the given value
+     * @throws NullPointerException if called with {@code null}
+     */
+    public void setBasementType(BasementType basementType) {
+        this.basementType = Objects.requireNonNull(basementType);
+    }
+
+    /**
+     * @return the current CF of this building (ie: with any damage applied immediately)
+     */
+    public int getCurrentCF() {
+        return currentCF;
+    }
+
+    /**
+     * Sets the current CF of this building to the given value
+     * @throws IllegalArgumentException if called with a negative value
+     */
+    public void setCurrentCF(int currentCF) {
+        this.currentCF = requireValidCF(currentCF);
+    }
+
+    /**
+     * @return the CF this building had at the start of the last phase
+     */
+    public int getPhaseCF() {
+        return phaseCF;
+    }
+
+    /**
+     * Sets the CF this building had at the start of the last phase to to the given value
+     * @throws IllegalArgumentException if called with a negative value
+     */
+    public void setPhaseCF(int phaseCF) {
+        this.phaseCF = requireValidCF(phaseCF);
+    }
+
+    /**
+     * @return the armor of this building
+     */
+    public int getArmor() {
+        return armor;
+    }
+
+    /**
+     * Sets the armor of this building to the given value
+     * @throws IllegalArgumentException if called with a negative value
+     */
+    public void setArmor(int armor) {
+        this.armor = requireValidArmor(armor);
+    }
+
+    /**
+     * @return whether the basement of this building section is collapsed
+     */
+    public boolean isBasementCollapsed() {
+        return basementCollapsed;
+    }
+
+    /**
+     * Sets whether the basement of this building section is collapsed
+     */
+    public void setBasementCollapsed(boolean basementCollapsed) {
+        this.basementCollapsed = basementCollapsed;
+    }
+
+    /**
+     * @return whether this building section is burning
+     */
+    public boolean isBurning() {
+        return burning;
+    }
+
+    /**
+     * Sets whether this building section is burning
+     */
+    public void setBurning(boolean burning) {
+        this.burning = burning;
+    }
+
+    /**
+     * @return a stream of the demolition charges planted in this building section
+     */
+    public Stream<DemolitionCharge> streamDemolitionCharges() {
```

Comment:
Provide a way to iterate.

Reply:
I don't get the point here, sorry.
What do you mean?

Reply:
Sorry, you're right. That's a half-completed thought that doesn't make any sense. What I meant to say is provide a way to iterate using a loop. Previously it returned the list directly, which is not usually the best approach, but it allowed processing by loops or streams. You have restricted access to streams only, which makes it harder to step through code. And some people don't like streams and would much rather use loops.

Reply:
That goes for streamSections as well.

Reply:
Oh, I see your point there.

The reason why I'm more at ease with exposing stuff as streams rather than collections/iterables/iterators/arrays is that streams are read-only.

With collections, people often wrap via `Collections.unmodifiableXXX` before returning, and arrays are (should be) treated as copy-on-read entities – streams don't require any of this craziness and one can just `.collect(toList())` or `.toArray()` if he wants to use a different API.

Personally, I advocate for using streams whenever possible, but... (or course) it is your call when it comes to MM! :-) Just let me know if you want me to use collections instead of streams (I guess providing both as alternatives wouldn't make much sense) – no big deal.

Reply:
I thought that was probably the reason. I tend to use streams frequently within the body of a method, but when it comes to public accessors I usually wrap a collection in an unmodifiable, which is not a perfect solution but it's a lightweight wrapper that prevents modification of the collection itself and easily lends itself to iteration by loop or stream without the overhead of converting a `Stream` to an `Iterable` or an `Iterator` to a `Stream`.